contracts/DependencyRegistryV1.sol

```diff
@@ -0,0 +1,1074 @@
+    }
+
+    /**
+     * @notice Converts bytes to hexadecimal string representation.
+     * https://ethereum.stackexchange.com/a/126928
+     * @param buffer bytes to be converted.
+     * @return string representation of buffer.
+     */
+    function bytesToHexString(
+        bytes memory buffer
+    ) public pure returns (string memory) {
+        // Fixed buffer size for hexadecimal convertion
+        bytes memory converted = new bytes(buffer.length * 2);
+
+        bytes memory _base = "0123456789abcdef";
+
+        for (uint256 i = 0; i < buffer.length; i++) {
+            converted[i * 2] = _base[uint8(buffer[i]) / _base.length];
+            converted[i * 2 + 1] = _base[uint8(buffer[i]) % _base.length];
+        }
+
+        return string(abi.encodePacked("0x", converted));
+    }
+
+    function getTokenHtml(
+        address _contractAddress,
+        uint256 _tokenId
+    ) external view returns (string memory) {
+        uint256 ONE_MILLION = 1_000_000;
+
+        uint256 projectId = _tokenId / ONE_MILLION;
+        bytes32 dependencyType = stringToBytes32(
+            this.getDependencyTypeForProject(_contractAddress, projectId)
+        );
+        bytes32 tokenHash = IDependencyRegistryCompatibleV0(_contractAddress)
+            .tokenIdToHash(_tokenId);
+
+        WrappedScriptRequest[] memory requests = new WrappedScriptRequest[](5);
+
+        requests[0]
+            .scriptContent = 'let css="html{height:100%}body{min-height:100%;margin:0;padding:0}canvas{padding:0;margin:auto;display:block;position:absolute;top:0;bottom:0;left:0;right:0}",head=document.head,style=document.createElement("style");head.appendChild(style),style.type="text/css",style.appendChild(document.createTextNode(css));';
+
+        requests[1].scriptContent = abi.encodePacked(
+            'let tokenData = {"tokenId":"',
+            Strings.toString(_tokenId),
+            '"',
+            ',"hash":"',
+            bytesToHexString(abi.encodePacked(tokenHash)),
+            '"}'
+        );
+
+        requests[2].name = "dependency";
+        requests[2].contractData = abi.encode(dependencyType);
+        requests[2].contractAddress = address(this);
+        requests[2].wrapType = 2; // <script type="text/javascript+gzip" src="data:text/javascript;base64,[script]"></script>
+
+        requests[3].name = "gunzipScripts-0.0.1.js";
+        requests[3].wrapType = 1; // <script src="data:text/javascript;base64,[script]"></script>
+        requests[3].contractAddress = ethFsAddress;
+
+        requests[4].name = "project";
+        requests[4].contractData = abi.encode(_contractAddress, projectId);
+        requests[4].contractAddress = address(this);
+        requests[4].wrapType = 0; // <script>[script]</script>
+
+        uint256 bufferSize = scriptyBuilder.getBufferSizeForEncodedHTMLWrapped(
+            requests
+        );
+
+        bytes memory base64EncodedHTMLDataURI = scriptyBuilder
+            .getEncodedHTMLWrapped(requests, bufferSize);
+
```

Comment:
Regarding having base64/raw HTML returning methods. 

If you want to support both versions you can use scripty's these methods together:
- getEncodedHTMLWrapped() -> bytes | base64 data uri 
- getHTMLWrapped() -> bytes | raw html

Maybe an internal method can return the requests + bufferSize and then you can have 2 separate methods that can return raw HTML and base64 encoded one. scripty internally calculates the buffer size offset for base64 conversion so you can use same requests and bufferSize for both methods.